engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetViewportToSizeOf.java

```diff
@@ -15,79 +15,59 @@
  */
 package org.terasology.rendering.dag.stateChanges;
 
-import org.terasology.assets.ResourceUrn;
 import org.terasology.registry.CoreRegistry;
-import org.terasology.rendering.opengl.BaseFBOsManager;
-import org.terasology.rendering.opengl.FBOManagerSubscriber;
-import java.util.Objects;
 import org.terasology.rendering.dag.StateChange;
 import org.terasology.rendering.opengl.FBO;
 import org.terasology.rendering.opengl.fbms.DisplayResolutionDependentFBOs;
 
-import static org.terasology.rendering.opengl.fbms.DisplayResolutionDependentFBOs.READONLY_GBUFFER;
+import java.util.Objects;
+
 import static org.lwjgl.opengl.GL11.glViewport;
 
 /**
  * TODO: Add javadocs
  */
-public final class SetViewportToSizeOf implements FBOManagerSubscriber, StateChange {
+public final class SetViewportToSizeOf implements StateChange {
     private static SetViewportToSizeOf defaultInstance;
 
-    private BaseFBOsManager fboManager;
-    private ResourceUrn fboName;
-    private int fboWidth;
-    private int fboHeight;
-    
-    @SuppressWarnings("FieldCanBeLocal")
     private FBO fbo;
 
     /**
      * The constructor, to be used in the initialise method of a node.
      *
      * Sample use:
-     *      addDesiredStateChange(new SetViewportToSizeOf("engine:sceneOpaque", displayResolutionDependentFboManager);
+     *      addDesiredStateChange(new SetViewportToSizeOf(fbo);
      *
-     * @param fboName a URN identifying an FBO.
-     * @param frameBuffersManager the BaseFBOsManager instance that will send change notifications via the update() method of this class.
+     * @param fbo The FBO whose dimensions the viewport will be resized to.
      */
-    public SetViewportToSizeOf(ResourceUrn fboName, BaseFBOsManager frameBuffersManager) {
-        this.fboManager = frameBuffersManager;
-        this.fboName = fboName;
-
-        update(); // Cheeky way to initialise fbo, fboWidth, fboHeight
-        fboManager.subscribe(this);
+    public SetViewportToSizeOf(FBO fbo) {
+        this.fbo = fbo;
     }
 
     @Override
     public StateChange getDefaultInstance() {
         if (defaultInstance == null) {
-            defaultInstance = new SetViewportToSizeOf(READONLY_GBUFFER, CoreRegistry.get(DisplayResolutionDependentFBOs.class));
+            DisplayResolutionDependentFBOs displayResolutionDependentFBOs = CoreRegistry.get(DisplayResolutionDependentFBOs.class);
+            assert(displayResolutionDependentFBOs != null);
```

Comment:
I mostly did this to avoid warnings by IntelliJ.

However, now that I think about it, the warning is a sign of (possibly) bad code, and shouldn't be covered up.

I'll revert this!